api/client/proxy/client.go

```diff
@@ -115,6 +118,22 @@ func (c *ClientConfig) CheckAndSetDefaults() error {
 			// not to send the client certificate by looking at certificate request.
 			if len(tlsCfg.Certificates) > 0 {
 				cert := tlsCfg.Certificates[0]
+
+				// When a hardware key is used to store the private key, the user may fail to provide
+				// a PIN or touch before a gRPC dial timeout occurs.
+				// The resulting "dial timeout" error is generic and doesn't indicate an issue with the
+				// hardware key itself (since YubiKey is treated like any other key).
+				// To avoid this, we perform a "warm-up" call to the key, ensuring it is ready
+				// before initiating the gRPC dial.
+				// This approach works because the connection is cached for a few seconds,
+				// allowing subsequent calls without requiring additional user action.
+				if priv, ok := cert.PrivateKey.(*keys.YubiKeyPrivateKey); ok {
```

Comment:
Ah, I can't assert to `keys.YubiKeyPrivateKey` because it lives in `yubikey.go` which is built only if `piv` tag is set.
Is this okay to move just this single struct to `yubikey_common.go` or not really?
Btw originally I tried `priv, ok := cert.PrivateKey.(*keys.PrivateKey); ok && priv.IsHardware()` but it didn't work :/

Reply:
You can't move the struct because 2 of the fields are types from `github.com/go-piv/piv-go/piv` which requires build support IIUC

Reply:
perhaps you could make YubiKeyPrivateKey implement a func `WarmupHardwareKey(context.Context) error`, and here check if cert.PrivateKey implements that interface

Reply:
That's a great idea, thanks! It's actually better than what I had before, because I need to call the private `y.sign` (to pass the context).